cmd/bom/cmd/validate.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"os"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/bom/pkg/spdx"
+	"sigs.k8s.io/release-utils/util"
+)
+
+func AddValidate(parent *cobra.Command) {
+	valOpts := validateOptions{
+		files: []string{},
+	}
+
+	cmd := &cobra.Command{
+		Short: "bom validate → Check artifacts against an sbom",
+		Long: `bom validate → Check artifacts against an sbom
+
+validate is the bom subcommand to check artifacts against SPDX
+manifests.
+
+This is an experimental command. The first iteration has support
+for checking files.
+
+`,
+		Use:               "validate",
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+		PersistentPreRunE: initLogging,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			for i, arg := range args {
+				if util.Exists(arg) {
+					file, err := os.Open(arg)
+					if err != nil {
+						return errors.Wrapf(err, "checking argument %d", i)
+					}
+					defer file.Close()
+					fileInfo, err := file.Stat()
+					if err != nil {
+						return errors.Wrapf(err, "calling stat on argument %d", i)
+					}
+					if fileInfo.IsDir() {
+						return errors.Errorf(
+							"the path %s is a directory, only files are supported at this time",
+							file.Name(),
+						)
+					}
+					if i == 0 {
+						valOpts.sbomPath = arg
+					} else {
+						valOpts.files = append(valOpts.files, file.Name())
+					}
+				} else {
+					return errors.Errorf("the path specified at %s does not exist", arg)
+				}
+			}
+			return validateArtifacts(valOpts)
+		},
+	}
+
+	cmd.PersistentFlags().StringSliceVarP(
+		&valOpts.files,
+		"files",
+		"f",
+		[]string{},
+		"list of files to verify",
+	)
+
+	cmd.PersistentFlags().BoolVarP(
+		&valOpts.exitCode,
+		"exit-code",
+		"e",
+		false,
+		"when true, bom will exit with exit code 1 if invalid artifacts are found",
+	)
```

Comment:
I think this is maybe extraneous?
Is there a scenario where we'd want a `validate` command to "fail successfully"?

Reply:
Yes! The idea is that you check the integrity of files you download and use the tool in a script:

`bom validate -e sbom.spdx file.zip || echo "Your zipfile is corrupt"`  

The bom action will use it to verify bom itself:

https://github.com/puerco/bom-installer/blob/1d65808b8b512ca77cf2d73cb844570cfec9f39e/action.yaml#L161